### ISSUE TITLE
Authorize files upload only in a specific subfolder of shared public folder

### ISSUE THREAD

**Issue by Oreonan** [nextcloud/server#20793]
**Is your feature request related to a problem? Please describe.**
When a folder is share with public link, we have choice to allow only download, only upload or download and upload, but we don't have ability to specific only a specific subfolder when upload is authorize instead of complete folder

**Describe the solution you'd like**
Add an option to allow upload in a particular subfolder instead of just on all folder and subfolders

**Describe alternatives you've considered**

**Additional context**

**Comment by jospoortvliet** [comment]
I'm guessing this would have a performance impact (recursive permissions, which is why we only have those on groupfolders).

But more importantly, I think it would create a bad user experience, complicated sharing significantly and risk user mistakes. For example: how do you see that a subfolder of the folder share you are sharing has write access? That's a security risk right there!

Here is our current folder share view:
![share link folder](https://user-images.githubusercontent.com/551757/80924170-c1381180-8d87-11ea-81c8-8cd5c04a2709.png)

You would have to find a clear way to:
1. choose a subfolder that has write access
2. make VERY clear in that dialog that a subfolder has gotten write access

I think the work-around of just creating a link share to a sub folder with write permissions has far less risk of causing mistakes and works just fine. I really don't see the added benefit here.

**Comment by martin-77** [comment]
Did not find a better issue, I think it fits somehow...

**The Setting:**
I use a folder-structure like this:

Main-Folder
   -Subfolder1
        -Sub-Subfolder1
        -Sub-Subfodler2
        -Sub-Subfolder3
   -Subfolder2
        -Sub-Subfolder1
        -Sub-Subfodler2
        -Sub-Subfolder3
   -Subfolder3
        -Sub-Subfolder1
        -Sub-Subfodler2
        -Sub-Subfolder3

Titles of Subfolder 1-3 are **different**.
Titles of Sub-Subfolder1-3 are the **same**.

My users are allowed to view Subfolders1 to Subfolders3
Some users must have the privilege to edit files inside Sub-Subfolder2

As a work-around, I made a group of persons, to whom I gave write access to Sub-Subfolder2 and shared this folder to that group.
The same users are part of the group, which have read-access to Subfolder1-3

**The Problem:** 
The usergroup with write-access see the Sub-Subfolder2 under their Main-Folder. In consequence, they have no idea, to which topic they are working. (which is defined by the title of Subfolder1-3) 

**My whish:**
It would be nice, if I could give the right to edit a file/folder inside the Sub-Subfolder an keep the given Folder-Structure.
